test/e2e/nomad/e2e_test.go

```diff
@@ -0,0 +1,144 @@
+//go:build e2e
+
+package e2e
+
+import (
+	"context"
+	"fmt"
+	"os"
+	"strconv"
+	"testing"
+
+	nomad "github.com/hashicorp/nomad/api"
+	"github.com/hetznercloud/csi-driver/internal/driver"
+	"github.com/stretchr/testify/assert"
+)
+
+const (
+    initialCapacity = 10737418240 // 10GiB
+    resizedCapacity = 11811160064 // 11GiB
+    resizedCapacityGB = 11
+)
+
+var cluster *Cluster
+
+func TestMain(m *testing.M) {
+	var err error
+	cluster, err = NewCluster()
+	if err != nil {
+		fmt.Printf("%v\n", err)
+		os.Exit(1)
+	}
+
+	exitCode := m.Run()
+
+	if err := cluster.Cleanup(); err != nil {
+		fmt.Printf("%v\n", err)
+		os.Exit(1)
+	}
+
+	os.Exit(exitCode)
+}
+
+func TestGetPluginInfo(t *testing.T) {
+    plugin, _, err := cluster.nomadClient.CSIPlugins().Info(driver.PluginName, &nomad.QueryOptions{}) 
+    if err != nil {
+        t.Error(err)
+    }
+
+    assert.NotNil(t, plugin, "Expected plugin from Nomad to be not nil")
+
+    assert.Equalf(
+        t,
+        plugin.Version,
+        driver.PluginVersion,
+        "Expected plugin version %s, but got %s",
+        driver.PluginVersion,
+        plugin.Version,
+    )
+}
+
+func TestVolumeLifecycle(t *testing.T) {
```

Comment:
This looks reasonable for the first few tests, but just handles the controller part, as the volume is never actually used in a workload, it does not verify any of the topology and `node` parts.

In the future, I would also like to see a test that:

- deploys a job with the volume attached
- writes a file to the volume
- stop the job
- start another job on another server with the volume attached
- verify that the file exists

Reply:
Created such a test with an affinity to use another node.